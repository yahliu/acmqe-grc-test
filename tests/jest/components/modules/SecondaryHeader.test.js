/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
//curly braces means pure component without redux
//which is what we want in unit test
import { SecondaryHeader } from '../../../../src-web/components/modules/SecondaryHeader'
import { createMemoryHistory } from 'history'


const history = createMemoryHistory({
  'length':5,
  'action':'PUSH',
  'location':{
    'pathname':'/multicloud/policies/all',
    'search':'',
    'hash':''
  }
})

describe('SecondaryHeader component 1', () => {
  const location = {
    pathname: '/multicloud/policies'
  }
  it('renders as expected', () => {
    const component = renderer.create(
      <SecondaryHeader title='hello world' location={location} history={history} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 2', () => {
  const tabs = [{
      'id': 'grc-all',
      'label': 'tabs.grc.all',
      'url': '/multicloud/policies/all',
      'index': 0
    }]
  const location = {
    pathname: '/multicloud/policies/all'
  }
  it('renders clickable as expected', () => {
    const component = renderer.create(
      <SecondaryHeader
        title='hello world'
        tabs={tabs}
        location={location} history={history}
        links={[{
          'id':'create-policy',
          'label':'button.create.policy',
          'url':'/multicloud/policies/create'
        }]}
        userAccess={
          [
            {
              'namespace': 'calamari',
              'rules': {
                '*/*': [
                  '*'
                ]
              }
            },
          ]
        }
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders clickable as expected', () => {
    const component = renderer.create(
      <SecondaryHeader
        title='hello world'
        tabs={tabs}
        location={location} history={history}
        links={[{
          'id':'create-policy',
          'label':'button.create.policy',
          'url':'/multicloud/policies/create'
        }]}
        userAccess={
          [
            {
              'namespace': 'calamari',
              'rules': {
                'policy.open-cluster-management.io/policies': [
                  'get',
                  'list',
                  'watch',
                  'update',
                  'patch',
                  'create',
                  'delete',
                  'deletecollection'
                ]
              }
            },
          ]
        }
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders nonclickable as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        tabs={tabs}
        location={location} history={history}
        links={[{
          'id':'create-policy',
          'label':'button.create.policy',
          'url':'/multicloud/policies/create'
        }]}
        userAccess={
          [
            {
              'namespace': 'calamari',
              'rules': {
                'policy.open-cluster-management.io/policies': [
                  'get',
                  'list',
                  'watch',
                  'update',
                  'patch',
                ]
              }
            },
          ]
        }
      />
    )
    expect(component.instance().render()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 3', () => {
  const tabs = [
    {
      id: 'logs-tab2',
      label: 'tabs.grc.all',
      url: '/hello',
    }
  ]
  const location = {
    pathname: '/multicloud/policies/all'
  }
  it('renders as expected', () => {
    const component = renderer.create(
      <SecondaryHeader
        title='hello world'
        // eslint-disable-next-line jsx-a11y/aria-role
        tabs={tabs}
        location={location}
        history={history}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 4', () => {
  const location = {
    pathname: '/multicloud/policies/all'
  }
  it('renders as expected', () => {
    const component = renderer.create(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('SecondaryHeader component 5', () => {
  const location = {
    pathname: '/multicloud/policies/all'
  }
  it('clickTab as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history} />
    )
    expect(component.instance().clickTab('www.ibm.com')).toEqual('www.ibm.com')
  })
  it('renderHeader with description', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        description={{action:jest.fn(), display:jest.fn()}}
        location={location}
        history={history} />
    )
    expect(component.instance().renderHeader()).toMatchSnapshot()
  })
  it('renderHeader without description', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history} />
    )
    expect(component.instance().renderHeader()).toMatchSnapshot()
  })
  it('renderLinks nonclickable as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        links={[{
          'id':'create-policy',
          'label':'button.create.policy',
          'url':'/multicloud/policies/create'
        }]}
      />
    )
    expect(component.instance().renderLinks()).toMatchSnapshot()
  })
  it('renderTabs as expected', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        tabs={[
          {
            'id':'grc-all',
            'label':'tabs.grc.all',
            'url':'/multicloud/policies/all',
            'index':1
          },
        ]}
      />
    )
    expect(component.instance().renderTabs()).toMatchSnapshot()
  })

  it('renderBreadCrumb should return undefined given no breadcrumbItems', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
      />
    )
    expect(component.instance().renderBreadCrumb()).toBeUndefined()
  })
  it('renderBreadCrumb should return an empty array given an empty array of breadcrumbItems', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        breadcrumbItems={[]}
      />
    )
    expect(component.instance().renderBreadCrumb()).toMatchSnapshot()
  })
  it('renderBreadCrumb should return an array of links given an array of breadcrumbItems', () => {
    const component = shallow(
      <SecondaryHeader
        title='hello world'
        location={location}
        history={history}
        breadcrumbItems={[
          {
            id:'id1',
            label:'label1',
            url:'www.test1.com',
            handleClick:jest.fn(), noLocale:true
          },
          {
            id:'id2',
            label:'label2',
            url:'www.test2.com',
            noLocale:true
          },
          {
            id:'id3',
            label:'label3',
            url:'www.test3.com',
            handleClick:jest.fn(),
            noLocale:false
          },
          {
            id:'id4',
            label:'label4',
            url:'www.test4.com',
            noLocale:false
          }]}
      />
    )
    expect(component.instance().renderBreadCrumb()).toMatchSnapshot()
  })
})
