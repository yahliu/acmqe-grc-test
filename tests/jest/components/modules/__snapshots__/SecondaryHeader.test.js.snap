// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHeader component 1 renders as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper no-tabs no-breadcrumb no-buttons"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container no-breadcrumb"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
      </div>
    </header>
  </div>
</div>
`;

exports[`SecondaryHeader component 2 renders clickable as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper no-breadcrumb"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container no-breadcrumb"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <nav
          aria-label="hello world undefined"
          className="bx--tabs"
          role="navigation"
          selected={0}
        >
          <div
            aria-label="listbox"
            className="bx--tabs-trigger"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <a
              className="bx--tabs-trigger-text"
              href="#"
              onClick={[Function]}
              tabIndex={-1}
            >
              Policies
            </a>
            <svg
              alt="show menu options"
              aria-label="show menu options"
              fillRule="evenodd"
              height="5"
              role="img"
              viewBox="0 0 10 5"
              width="10"
            >
              <title>
                show menu options
              </title>
              <path
                d="M0 0l5 4.998L10 0z"
              />
            </svg>
          </div>
          <ul
            className="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              className="bx--tabs__nav-item bx--tabs__nav-item--selected"
              id="grc-all"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={true}
              tabIndex={-1}
            >
              <a
                aria-selected={true}
                className="bx--tabs__nav-link"
                href="/multicloud/policies/all"
                role="tab"
                tabIndex={0}
              >
                Policies
              </a>
            </li>
          </ul>
        </nav>
        <div
          aria-hidden={false}
          hidden={false}
          selected={true}
        />
      </div>
    </header>
  </div>
  <div
    className="secondary-header-links"
  >
    <button
      aria-disabled={false}
      aria-label={null}
      className="pf-c-button pf-m-primary pf-m-small"
      data-ouia-component-id="OUIA-Generated-Button-primary-1"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe={true}
      disabled={false}
      id="create-policy"
      onClick={[Function]}
      role={null}
      type="button"
    >
      Create policy
    </button>
  </div>
</div>
`;

exports[`SecondaryHeader component 2 renders clickable as expected 2`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper no-breadcrumb"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container no-breadcrumb"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <nav
          aria-label="hello world undefined"
          className="bx--tabs"
          role="navigation"
          selected={0}
        >
          <div
            aria-label="listbox"
            className="bx--tabs-trigger"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <a
              className="bx--tabs-trigger-text"
              href="#"
              onClick={[Function]}
              tabIndex={-1}
            >
              Policies
            </a>
            <svg
              alt="show menu options"
              aria-label="show menu options"
              fillRule="evenodd"
              height="5"
              role="img"
              viewBox="0 0 10 5"
              width="10"
            >
              <title>
                show menu options
              </title>
              <path
                d="M0 0l5 4.998L10 0z"
              />
            </svg>
          </div>
          <ul
            className="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              className="bx--tabs__nav-item bx--tabs__nav-item--selected"
              id="grc-all"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={true}
              tabIndex={-1}
            >
              <a
                aria-selected={true}
                className="bx--tabs__nav-link"
                href="/multicloud/policies/all"
                role="tab"
                tabIndex={0}
              >
                Policies
              </a>
            </li>
          </ul>
        </nav>
        <div
          aria-hidden={false}
          hidden={false}
          selected={true}
        />
      </div>
    </header>
  </div>
  <div
    className="secondary-header-links"
  >
    <button
      aria-disabled={false}
      aria-label={null}
      className="pf-c-button pf-m-primary pf-m-small"
      data-ouia-component-id="OUIA-Generated-Button-primary-2"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe={true}
      disabled={false}
      id="create-policy"
      onClick={[Function]}
      role={null}
      type="button"
    >
      Create policy
    </button>
  </div>
</div>
`;

exports[`SecondaryHeader component 2 renders nonclickable as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper no-breadcrumb"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container no-breadcrumb"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <Tabs
          aria-label="hello world undefined"
          ariaLabel="listbox"
          iconDescription="show menu options"
          role="navigation"
          selected={0}
          triggerHref="#"
        >
          <Tab
            href="/multicloud/policies/all"
            id="grc-all"
            label="Policies"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="presentation"
            selected={false}
            tabIndex={0}
          />
        </Tabs>
      </div>
    </header>
  </div>
  <div
    className="secondary-header-links"
  >
    <mockTooltip
      content="You are not authorized to complete this action. See your cluster administrator for role-based access information."
      zIndex={10001}
    >
      <div
        className="disable-tooltip"
      >
        <Button
          id="create-policy"
          isDisabled={true}
          isSmall={true}
          onClick={[Function]}
          variant="primary"
        >
          Create policy
        </Button>
      </div>
    </mockTooltip>
  </div>
</div>
`;

exports[`SecondaryHeader component 3 renders as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper no-breadcrumb no-buttons"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container no-breadcrumb"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
        <nav
          aria-label="hello world undefined"
          className="bx--tabs"
          role="navigation"
          selected={0}
        >
          <div
            aria-label="listbox"
            className="bx--tabs-trigger"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <a
              className="bx--tabs-trigger-text"
              href="#"
              onClick={[Function]}
              tabIndex={-1}
            >
              Policies
            </a>
            <svg
              alt="show menu options"
              aria-label="show menu options"
              fillRule="evenodd"
              height="5"
              role="img"
              viewBox="0 0 10 5"
              width="10"
            >
              <title>
                show menu options
              </title>
              <path
                d="M0 0l5 4.998L10 0z"
              />
            </svg>
          </div>
          <ul
            className="bx--tabs__nav bx--tabs__nav--hidden"
            role="tablist"
          >
            <li
              className="bx--tabs__nav-item bx--tabs__nav-item--selected"
              id="logs-tab2"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={true}
              tabIndex={-1}
            >
              <a
                aria-selected={true}
                className="bx--tabs__nav-link"
                href="/hello"
                role="tab"
                tabIndex={0}
              >
                Policies
              </a>
            </li>
          </ul>
        </nav>
        <div
          aria-hidden={false}
          hidden={false}
          selected={true}
        />
      </div>
    </header>
  </div>
</div>
`;

exports[`SecondaryHeader component 4 renders as expected 1`] = `
<div
  aria-label="hello world"
  className="secondary-header-wrapper no-tabs no-breadcrumb no-buttons"
  role="region"
>
  <div
    className="secondary-header secondary-header-grc-overview simple-header"
  >
    <header
      aria-label="Heading: hello world"
    >
      <div
        className="bx--detail-page-header-content"
      >
        <div
          className="bx--detail-page-header-title-container no-breadcrumb"
        >
          <h1
            className="bx--detail-page-header-title"
          >
            hello world
          </h1>
        </div>
      </div>
    </header>
  </div>
</div>
`;

exports[`SecondaryHeader component 5 renderBreadCrumb should return an array of links given an array of breadcrumbItems 1`] = `
Array [
  <div
    className="bx--breadcrumb-item"
    title="label1"
  >
    <div
      className="breadcrumb-link"
      onClick={[MockFunction]}
    />
  </div>,
  <div
    className="bx--breadcrumb-item"
    title="label2"
  >
    <Link
      className="bx--link"
      to="www.test2.com"
    >
      label2
    </Link>
  </div>,
  <div
    className="bx--breadcrumb-item"
    title="label3"
  >
    <div
      className="breadcrumb-link"
      onClick={[MockFunction]}
    />
  </div>,
  <div
    className="bx--breadcrumb-item"
    title="label4"
  >
    <Link
      className="bx--link"
      to="www.test4.com"
    />
  </div>,
]
`;

exports[`SecondaryHeader component 5 renderBreadCrumb should return an empty array given an empty array of breadcrumbItems 1`] = `Array []`;

exports[`SecondaryHeader component 5 renderHeader with description 1`] = `
<div
  className="bx--detail-page-header-title-container no-breadcrumb"
>
  <h1
    className="bx--detail-page-header-title"
  >
    hello world
  </h1>
  <div
    className="detail-page-header-title-button"
    onClick={[MockFunction]}
  >
    <p>
      [MockFunction]
    </p>
  </div>
</div>
`;

exports[`SecondaryHeader component 5 renderHeader without description 1`] = `
<div
  className="bx--detail-page-header-title-container no-breadcrumb"
>
  <h1
    className="bx--detail-page-header-title"
  >
    hello world
  </h1>
</div>
`;

exports[`SecondaryHeader component 5 renderLinks nonclickable as expected 1`] = `
Array [
  <mockTooltip
    content="You are not authorized to complete this action. See your cluster administrator for role-based access information."
    zIndex={10001}
  >
    <div
      className="disable-tooltip"
    >
      <Button
        id="create-policy"
        isDisabled={true}
        isSmall={true}
        onClick={[Function]}
        variant="primary"
      >
        Create policy
      </Button>
    </div>
  </mockTooltip>,
]
`;

exports[`SecondaryHeader component 5 renderTabs as expected 1`] = `
Array [
  <Tab
    href="/multicloud/policies/all"
    id="grc-all"
    label="Policies"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    selected={false}
    tabIndex={0}
  />,
]
`;
